File: debugModule/src/main/libraryKotlin/java/com/component/debug/librarykotlin/LibraryKotlinMainActivity.java
package com.component.debug.librarykotlin;


import android.os.Bundle;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import com.component.debug.R;
import com.plugin.component.ComponentManager;
import com.plugin.component.SdkManager;
import com.plugin.librarykotlin.IProvideFromKotlin;

public class LibraryKotlinMainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_debug_layout);
        ComponentManager.init(getApplication());
        ((TextView) findViewById(R.id.text)).setText(SdkManager.getSdk(IProvideFromKotlin.class).provideString());
    }
}

File: librarykotlin/src/main/debug/java/com/plugin/librarykotlin/LibraryKotlinMainActivity.kt
package com.plugin.librarykotlin

import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.plugin.component.ComponentManager
import com.plugin.component.SdkManager
import com.plugin.library.ISdk

import com.plugin.librarykotlin.R

class LibraryKotlinMainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.main_activity_layout)
        ComponentManager.init(application)
        Toast.makeText(this, SdkManager.getSdk(ISdk::class.java)!!.sdkName, Toast.LENGTH_LONG).show()
    }
}
